File: SELF-Backend/SELF.Application/Dashboard/DashboardService.cs
using Microsoft.EntityFrameworkCore;
using SELF.Application.Dashboard.DTOs;
using SELF.Application.Dashboard.Interfaces;
using SELF.Infrastructure.Data;

namespace SELF.Application.Dashboard;

public class DashboardService(ApplicationDbContext context) : IDashboardService
{
    private readonly ApplicationDbContext _context = context;

    public async Task<DashboardStatsResponse> GetDashboardStatsAsync(string? year = null, string? state = null)
    {
        var totalNgos = await _context.NGOs.CountAsync(n => !n.IsDeleted);
        var totalApps = await _context.Applications.CountAsync(a => !a.IsDeleted);
        var totalJobs = await _context.Jobs.CountAsync(j => !j.IsDeleted && j.IsActive);

        var stats = new DashboardStatsResponse
        {
            TotalNGOs = totalNgos > 0 ? totalNgos : 4280,
            ActiveProjects = totalApps > 0 ? totalApps : 1240,
            TotalGrantsSanctionedCr = 185.4m,
            TotalGrantsDisbursedCr = 162.8m,
            TotalBeneficiaries = 425000,
            TotalJobs = totalJobs > 0 ? totalJobs : 12,
            TotalApplications = totalApps > 0 ? totalApps : 3150,
            MonthlyTrends =
            [
                new() { Month = "Jan", Amount = 12.4m, Projects = 45 },
                new() { Month = "Feb", Amount = 14.8m, Projects = 52 },
                new() { Month = "Mar", Amount = 24.5m, Projects = 89 },
                new() { Month = "Apr", Amount = 11.2m, Projects = 38 },
                new() { Month = "May", Amount = 15.6m, Projects = 49 },
                new() { Month = "Jun", Amount = 18.3m, Projects = 61 },
                new() { Month = "Jul", Amount = 19.8m, Projects = 67 },
                new() { Month = "Aug", Amount = 22.1m, Projects = 74 }
            ],
            SectorDistribution =
            [
                new() { Sector = "Health & Nutrition", Count = 320, Percentage = 32.0m },
                new() { Sector = "Education & Skill Dev", Count = 280, Percentage = 28.0m },
                new() { Sector = "Agriculture & Livelihoods", Count = 210, Percentage = 21.0m },
                new() { Sector = "Environment & Solar", Count = 110, Percentage = 11.0m },
                new() { Sector = "Disability & Inclusion", Count = 80, Percentage = 8.0m }
            ],
            StateWiseDistribution =
            [
                new() { State = "Jharkhand", Projects = 480, GrantsCr = 62.4m },
                new() { State = "Odisha", Projects = 310, GrantsCr = 41.2m },
                new() { State = "Chhattisgarh", Projects = 260, GrantsCr = 35.8m },
                new() { State = "Madhya Pradesh", Projects = 190, GrantsCr = 28.0m },
                new() { State = "Bihar", Projects = 140, GrantsCr = 18.0m }
            ]
        };

        return stats;
    }
}

File: self-main/src/services/dashboardService.ts
import { api } from './api';
import type { ApiResponse } from './careerService';

export interface MonthlyGrantTrend {
  month: string;
  amount: number;
  projects: number;
}

export interface SectorDistribution {
  sector: string;
  count: number;
  percentage: number;
  color?: string;
}

export interface StateWiseProject {
  state: string;
  projects: number;
  grantsCr: number;
}

export interface DashboardStats {
  totalNGOs: number;
  activeProjects: number;
  totalGrantsSanctionedCr: number;
  totalGrantsDisbursedCr: number;
  totalBeneficiaries: number;
  totalJobs: number;
  totalApplications: number;
  monthlyTrends: MonthlyGrantTrend[];
  sectorDistribution: SectorDistribution[];
  stateWiseDistribution: StateWiseProject[];
}

export const dashboardService = {
  getStats: async (year?: string, state?: string): Promise<DashboardStats> => {
    const response = await api.get<ApiResponse<DashboardStats>>('/Dashboard/stats', {
      params: { year, state },
    });
    if (!response.data.success || !response.data.data) {
      throw new Error(response.data.message || 'Failed to fetch dashboard statistics.');
    }
    return response.data.data;
  },
};
